### PR TITLE
feat: job summary

### PR DESCRIPTION
**Pull request by gentlementlegen**
QA: https://github.com/Meniole/automated-merging/actions/runs/10484217034

This adds a summary on the job which is very useful to see what got merged or not without having to go through the logs.

### CODE REVIEW

File: src/handlers/summary.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+function generateGitHubSummary(context: Context, urls: ResultInfo[]): string {
+  let monitored = `Monitored: [${context.config.repos.monitor.join(" | ")}]\n\n`;
+  monitored += `Ignored: [${context.config.repos.ignore.join(" | ")}]\n\n`;
+  monitored += urls
+    .map(({ url, merged }) => {
+      const status = merged ? `<span>🟢 merged</span>` : `<span>🔵 no change</span>`;
+      return `- [${url}](${url}) - ${status}`;
+    })
+    .join("\n");
```
Comment (0x4007) [inline src/handlers/summary.ts:17]: You should switch the order of the "columns" because the URLs are of variable length, where the indicators there are only two lengths. It will present more cleanly with the indicators first, and then the URLs after.

Comment (gentlementlegen) [reply src/handlers/summary.ts:17]: maybe it would be better to just do a table?

File: src/plugin.ts [hunk 2]
```diff
@@ -20,8 +20,6 @@ export async function plugin(inputs: PluginInputs, env: Env) {
     logger: new Logs("debug"),
   };
 
-  context.logger.info(
-    `Will check the following repos / orgs: [${context.config.repos.monitor.join(", ")}] and ignore [${context.config.repos.ignore.join(", ")}]`
-  );
+  context.logger.info(`Will check the following repos: [${context.config.repos.monitor.join(", ")}] and ignore [${context.config.repos.ignore.join(", ")}]`);
```
Comment (Keyrxng) [inline src/plugin.ts:23]: Is it maybe cleaner to log something like `("Will check the following repos:", {repos: config.repos})`? I mean cleaner/easier to read in the log itself

File: src/helpers/update-pull-requests.ts [hunk 3]
```diff
@@ -27,20 +28,30 @@ function isIssueEvent(event: object): event is IssueEvent {
...
+    }
   }
 
   const pullRequests = await getOpenPullRequests(context, context.config.repos);
 
   if (!pullRequests?.length) {
     return logger.info("Nothing to do.");
   }
+
+  let isMerged = false;
```
Comment (Keyrxng) [inline src/helpers/update-pull-requests.ts:48]: Since this is only used in the loop and reset on each iteration is there any reason not to define it just inside the loop?

**Comment by Keyrxng** [comment]
This is nice I hadn't considered outputting log summaries before

File: src/helpers/github.ts [hunk 4]
```diff
@@ -152,9 +147,14 @@ export async function getOpenPullRequests(context: Context, targets: ReposWatchS
       return acc;
     }, []),
   ];
+  if (!monitor.length) {
+    filter.push(`org:${context.payload.repository.owner?.login}`);
```
Comment (whilefoo) [inline src/helpers/github.ts:151]: this is not part of the PR but I was just thinking that its possible a repo is not owned by an org, so actually it should check the owner type and default to org or repo

**Review by whilefoo — APPROVED** [review]
looks great and helps to debug!

File: src/helpers/github.ts (hunk 4 above)
Comment (gentlementlegen) [reply src/helpers/github.ts:151]: @whilefoo It seems you can put a user name as an org and it works still:
https://github.com/search?q=org%3Awhilefoo+is%3Apr

File: src/handlers/summary.ts (hunk 1 above)
Comment (gentlementlegen) [reply src/handlers/summary.ts]: @0x4007 How about this
https://github.com/Meniole/automated-merging/actions/runs/10501574194/attempts/1#summary-29091781547

File: src/plugin.ts (hunk 2 above)
Comment (gentlementlegen) [reply src/plugin.ts]: https://github.com/Meniole/automated-merging/actions/runs/10501574194/job/29091781547#step:6:34

**Comment by 0x4007** [comment]
Still doing minor refinements but perhaps you can get started by adjusting to the following style:

# Preview

## Merge Report

<samp>

| Merged | ID |
|--------|-----|
| 🔵      | [assistive-pricing#3](https://github.com/Meniole/assistive-pricing/pull/3) |
| ⚫️     | [ubiquibot-kernel#5](https://github.com/Meniole/ubiquibot-kernel/pull/5) |
| ⚫️     | [command-wallet#7](https://github.com/Meniole/command-wallet/pull/7) |
| ⚫️     | [command-start-stop#4](https://github.com/Meniole/command-start-stop/pull/4) |
| ⚫️     | [test-automated-merging#1](https://github.com/Meniole/test-automated-merging/pull/1) |
| ⚫️     | [automated-merging#1](https://github.com/Meniole/automated-merging/pull/1) |
| ⚫️     | [command-wallet#1](https://github.com/Meniole/command-wallet/pull/1) |
| ⚫️     | [user-activity-watcher#3](https://github.com/Meniole/user-activity-watcher/pull/3) |
| ⚫️     | [command-query-user#3](https://github.com/Meniole/command-query-user/pull/3) |
| ⚫️     | [ts-template#6](https://github.com/Meniole/ts-template/pull/6) |

</samp>


## Configuration
### Watching Repositories
### Ignored Repositories
- [conversation-rewards](https://github.com/ubiquibot/conversation-rewards)

# Source Code

```
## Merge Report

<samp>

| Merged | ID |
|--------|-----|
| 🔵      | [assistive-pricing#3](https://github.com/Meniole/assistive-pricing/pull/3) |
| ⚫️     | [ubiquibot-kernel#5](https://github.com/Meniole/ubiquibot-kernel/pull/5) |
| ⚫️     | [command-wallet#7](https://github.com/Meniole/command-wallet/pull/7) |
| ⚫️     | [command-start-stop#4](https://github.com/Meniole/command-start-stop/pull/4) |
| ⚫️     | [test-automated-merging#1](https://github.com/Meniole/test-automated-merging/pull/1) |
| ⚫️     | [automated-merging#1](https://github.com/Meniole/automated-merging/pull/1) |
| ⚫️     | [command-wallet#1](https://github.com/Meniole/command-wallet/pull/1) |
| ⚫️     | [user-activity-watcher#3](https://github.com/Meniole/user-activity-watcher/pull/3) |
| ⚫️     | [command-query-user#3](https://github.com/Meniole/command-query-user/pull/3) |
| ⚫️     | [ts-template#6](https://github.com/Meniole/ts-template/pull/6) |

</samp>


## Configuration
### Watching Repositories
### Ignored Repositories
- [conversation-rewards](https://github.com/ubiquibot/conversation-rewards)

```

**Comment by gentlementlegen** [comment]
@0x4007 https://github.com/Meniole/automated-merging/actions/runs/10503093579

**Comment by 0x4007** [comment]
That's too bad i can't seem to see anything from mobile

**Comment by gentlementlegen** [comment]
image

**Comment by 0x4007** [comment]
I assume it's fine so we can merge.
